hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/onset/WorkflowOnset.java

```diff
@@ -15,16 +15,140 @@
  */
 package com.hedera.node.app.workflows.onset;
 
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TRANSACTION;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TRANSACTION_BODY;
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Parser;
 import com.hedera.node.app.SessionContext;
+import com.hedera.node.app.service.mono.exceptions.UnknownHederaFunctionality;
+import com.hedera.node.app.service.mono.utils.MiscUtils;
 import com.hedera.node.app.workflows.common.PreCheckException;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.SignatureMap;
+import com.hederahashgraph.api.proto.java.SignedTransaction;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
 
-// TODO: Implemented as part of ingest workflow, needs to be merged
+/**
+ * This class does some pre-processing before each workflow. It parses the provided {@link
+ * ByteBuffer} and checks it.
+ */
 public class WorkflowOnset {
```

Comment:
I assume this class is threadsafe? A single instance used for all workflows?

Reply:
Yes, this class is thread-safe. It can be used as a Singleton.